api/src/main/java/com/netflix/iceberg/encryption/EncryptionManager.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.netflix.iceberg.encryption;
+
+import com.netflix.iceberg.io.InputFile;
+import com.netflix.iceberg.io.OutputFile;
+
+import java.io.Serializable;
+import java.util.Iterator;
+
+/**
+ * Module for encrypting and decrypting table data files.
+ * <p>
+ * This must be serializable because an instance may be instantiated in one place and sent across
+ * the wire in some Iceberg integrations, notably Spark.
+ */
+public interface EncryptionManager extends Serializable {
+
+  /**
+   * Given an {@link EncryptedInputFile#encryptedInputFile()} representing the raw encrypted bytes
+   * from the underlying file system, and given metadata about how the file was encrypted via
+   * {@link EncryptedInputFile#keyMetadata()}, return an {@link InputFile} that returns decrypted
+   * input streams.
+   */
+  InputFile decrypt(EncryptedInputFile encrypted);
+
+  /**
+   * Variant of {@link #decrypt(EncryptedInputFile)} that provides a sequence of files that all
+   * need to be decrypted in a single context.
+   * <p>
+   * By default this calls the single-file decryption method for each element in the iterator.
+   * Implementations can override this for a variety of optimizations. For example, an
+   * implementation can perform lookahead on the input iterator and fetch encryption keys in batch.
+   */
+  default Iterator<InputFile> decrypt(Iterator<EncryptedInputFile> encrypted) {
```

Comment:
Sorry I missed this before. Why use an `Iterator` instead of an `Iterable`? I think that `Iterable` is usually a better API choice, unless there is some reason to restrict iteration to just one pass.